micrometer-observation-test/src/main/java/io/micrometer/observation/tck/TestObservationRegistry.java

```diff
@@ -122,7 +122,7 @@ public boolean equals(Object o) {
             if (this == o) {
                 return true;
             }
-            if (o == null || getClass() != o.getClass()) {
+            if (!(o instanceof TestObservationContext)) {
```

Comment:
I'm not sure if the previous logic was intentional. The `TestObservationContext` class is package private but non-final, so an argument could be made either way, but instanceof seems just as good even if it's probably not needed.

Reply:
I'm pretty sure that that was generated by Intellij

Reply:
In that case, no harm in changing it, I think. Thanks for the confirmation.